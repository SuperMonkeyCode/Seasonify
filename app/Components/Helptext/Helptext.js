/* @flow */
import { Content, Text } from 'native-base';
import { pure } from 'recompose';
import React from 'react';

import styles from './Helptext.style';

type Props = {
  /* parent */
  children: string | React$Element<*> | Array<*>,
  note?: boolean,
  fullscreen?: boolean,
  /* connect */
  /* HOC */
};

const enhance = pure;

function Helptext({ note = true, fullscreen = false, children }: Props) {
  return (
    <Content
      contentContainerStyle={fullscreen ? styles.fullscreenContainer : styles.container}
    >
      <Text note={note} style={fullscreen ? styles.fullscreenText : styles.text}>
        {children}
      </Text>
    </Content>
  );
}

export default enhance(Helptext);
