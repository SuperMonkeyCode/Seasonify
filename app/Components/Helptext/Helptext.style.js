import { fonts } from 'Themes';

export default {
  container: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'flex-start',
    alignItems: 'center',
  },
  fullscreenContainer: {
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
  },
  text: {
    fontStyle: 'italic',
    textAlign: 'left',
    fontSize: fonts.size.infos,
  },
  fullscreenText: {
    fontStyle: 'italic',
    paddingHorizontal: 40,
    textAlign: 'center',
    fontSize: fonts.size.infos,
  },
};
