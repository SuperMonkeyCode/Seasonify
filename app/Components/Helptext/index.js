/* @flow */
export { default } from './Helptext';
