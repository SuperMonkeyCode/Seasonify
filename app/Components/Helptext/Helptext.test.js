import React from 'react';

import Helptext from './Helptext';

function setup(specificProps = {}) {
  const props = {
    navigation: NavigationMock,
    ...specificProps,
  };
  const parent = shallow(<Helptext {...props} />);
  const level = [];
  level[0] = parent;
  level[1] = parent.dive();
  level[2] = level[1].dive();
  const component = parent;
  // const component = level[2];
  return {
    component,
    parent,
    level,
    props,
    // button: component.find(Button).first(),
  };
}

describe('Rendering when default state', () => {
  const { component } = setup();
  it('should match', () => {
    expect(component).toMatchSnapshot();
  });
});

describe('Rendering without note style and with fullscreen on', () => {
  const { component } = setup({ note: false, fullscreen: true });
  it('should match', () => {
    expect(component).toMatchSnapshot();
  });
});
