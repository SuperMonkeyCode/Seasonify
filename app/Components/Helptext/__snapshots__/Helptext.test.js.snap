// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering when default state should match 1`] = `
<Styled(Content)
  contentContainerStyle={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "flex-start",
    }
  }
>
  <Styled(Text)
    note={true}
    style={
      Object {
        "fontSize": 16,
        "fontStyle": "italic",
        "paddingHorizontal": 40,
        "textAlign": "center",
      }
    }
  />
</Styled(Content)>
`;

exports[`Rendering without note style and with fullscreen on should match 1`] = `
<Styled(Content)
  contentContainerStyle={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <Styled(Text)
    note={false}
    style={
      Object {
        "fontSize": 16,
        "fontStyle": "italic",
        "paddingHorizontal": 40,
        "textAlign": "center",
      }
    }
  />
</Styled(Content)>
`;
